In [1]:
#데이터 로드와 인덱스 번호 제거 
import pandas as pd
import numpy as np 
df = pd.read_csv('data/user_spec.csv')
df

application_id  user_id  birth_year  gender          insert_time  \
0               1249046   118218      1985.0     1.0  2022-06-07 06:28:18   
1                954900   553686      1968.0     1.0  2022-06-07 14:29:03   
2                137274    59516      1997.0     1.0  2022-06-07 21:40:22   
3               1570936   167320      1989.0     1.0  2022-06-07 09:40:27   
4                967833    33400      2000.0     1.0  2022-06-07 08:55:07   
...                 ...      ...         ...     ...                  ...   
1394211         1864587   489900      2000.0     1.0  2022-03-22 14:55:32   
1394212         1327066   151422      1955.0     1.0  2022-03-22 01:19:24   
1394213         1319606   173524      1983.0     1.0  2022-03-22 07:34:32   
1394214         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   
1394215          816537     3864      1977.0     0.0  2022-03-22 08:55:14   

         credit_score  yearly_income      income_type  company_enter_month  \
0               660.0    108000000.0  PRIVATEBUSINESS           20151101.0   
1               870.0     30000000.0  PRIVATEBUSINESS           20070201.0   
2               710.0     30000000.0       FREELANCER           20210901.0   
3               820.0     62000000.0     EARNEDINCOME           20170101.0   
4               630.0     36000000.0     EARNEDINCOME           20210901.0   
...               ...            ...              ...                  ...   
1394211         590.0     25000000.0       FREELANCER             202106.0   
1394212         980.0     20000000.0      OTHERINCOME                  NaN   
1394213         750.0     75000000.0     EARNEDINCOME             200908.0   
1394214         640.0     50000000.0     EARNEDINCOME             201705.0   
1394215           NaN     35000000.0       FREELANCER             201103.0   

        employment_type houseown_type  desired_amount purpose  \
0                    기타            자가       1000000.0      기타   
1                   정규직        기타가족소유      30000000.0    대환대출   
2                    기타        기타가족소유      10000000.0     생활비   
3                   정규직            자가       2000000.0     생활비   
4                   정규직        기타가족소유       5000000.0     생활비   
...                 ...           ...             ...     ...   
1394211              기타        기타가족소유       5000000.0    사업자금   
1394212              기타            자가      50000000.0     생활비   
1394213             정규직            자가     100000000.0    대환대출   
1394214             정규직            자가      10000000.0    대환대출   
1394215              기타            자가      20000000.0     생활비   

         personal_rehabilitation_yn  personal_rehabilitation_complete_yn  \
0                               0.0                                  NaN   
1                               0.0                                  NaN   
2                               0.0                                  NaN   
3                               0.0                                  NaN   
4                               0.0                                  0.0   
...                             ...                                  ...   
1394211                         NaN                                  NaN   
1394212                         NaN                                  NaN   
1394213                         NaN                                  NaN   
1394214                         NaN                                  NaN   
1394215                         NaN                                  NaN   

         existing_loan_cnt  existing_loan_amt  
0                      4.0        162000000.0  
1                      1.0         27000000.0  
2                      5.0         15000000.0  
3                      7.0        344000000.0  
4                      1.0         16000000.0  
...                    ...                ...  
1394211                NaN                NaN  
1394212                1.0                NaN  
1394213              

In [2]:
df.isnull().sum()


application_id                               0
user_id                                      0
birth_year                               12961
gender                                   12961
insert_time                                  0
credit_score                            105115
yearly_income                               90
income_type                                 85
company_enter_month                     171760
employment_type                             85
houseown_type                               85
desired_amount                              85
purpose                                     85
personal_rehabilitation_yn              587461
personal_rehabilitation_complete_yn    1203354
existing_loan_cnt                       198556
existing_loan_amt                       313774
dtype: int64

application_id                               0
user_id                                      0
birth_year                               12961 = 평균값
gender                                   12961 = 최빈
insert_time                                  0
credit_score                            105115 = 평균
yearly_income                               90 = 평균
income_type                                 85 = 최빈
company_enter_month                     171760 = 
employment_type                             85 = 최빈
houseown_type                               85 = 최빈
desired_amount                              85 = 평균
purpose                                     85 = 최빈
personal_rehabilitation_yn              587461 
personal_rehabilitation_complete_yn    1203354 
existing_loan\_cnt                       198556 = 0으로
existing_loan_amt                       313774 = 평균으로
dtype: int64

In [2]:
df1=df.dropna(subset=['credit_score'])
df1=df1.dropna(subset=['birth_year'])
df1=df1.dropna(subset=['income_type'])
df1=df1.dropna(subset=['gender'])
df1

application_id  user_id  birth_year  gender          insert_time  \
0               1249046   118218      1985.0     1.0  2022-06-07 06:28:18   
1                954900   553686      1968.0     1.0  2022-06-07 14:29:03   
2                137274    59516      1997.0     1.0  2022-06-07 21:40:22   
3               1570936   167320      1989.0     1.0  2022-06-07 09:40:27   
4                967833    33400      2000.0     1.0  2022-06-07 08:55:07   
...                 ...      ...         ...     ...                  ...   
1394210          316356    53701      1996.0     1.0  2022-03-22 10:01:38   
1394211         1864587   489900      2000.0     1.0  2022-03-22 14:55:32   
1394212         1327066   151422      1955.0     1.0  2022-03-22 01:19:24   
1394213         1319606   173524      1983.0     1.0  2022-03-22 07:34:32   
1394214         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   

         credit_score  yearly_income      income_type  company_enter_month  \
0               660.0    108000000.0  PRIVATEBUSINESS           20151101.0   
1               870.0     30000000.0  PRIVATEBUSINESS           20070201.0   
2               710.0     30000000.0       FREELANCER           20210901.0   
3               820.0     62000000.0     EARNEDINCOME           20170101.0   
4               630.0     36000000.0     EARNEDINCOME           20210901.0   
...               ...            ...              ...                  ...   
1394210         730.0     60000000.0  PRIVATEBUSINESS             202105.0   
1394211         590.0     25000000.0       FREELANCER             202106.0   
1394212         980.0     20000000.0      OTHERINCOME                  NaN   
1394213         750.0     75000000.0     EARNEDINCOME             200908.0   
1394214         640.0     50000000.0     EARNEDINCOME             201705.0   

        employment_type houseown_type  desired_amount purpose  \
0                    기타            자가       1000000.0      기타   
1                   정규직        기타가족소유      30000000.0    대환대출   
2                    기타        기타가족소유      10000000.0     생활비   
3                   정규직            자가       2000000.0     생활비   
4                   정규직        기타가족소유       5000000.0     생활비   
...                 ...           ...             ...     ...   
1394210             정규직           전월세      15000000.0    사업자금   
1394211              기타        기타가족소유       5000000.0    사업자금   
1394212              기타            자가      50000000.0     생활비   
1394213             정규직            자가     100000000.0    대환대출   
1394214             정규직            자가      10000000.0    대환대출   

         personal_rehabilitation_yn  personal_rehabilitation_complete_yn  \
0                               0.0                                  NaN   
1                               0.0                                  NaN   
2                               0.0                                  NaN   
3                               0.0                                  NaN   
4                               0.0                                  0.0   
...                             ...                                  ...   
1394210                         NaN                                  NaN   
1394211                         NaN                                  NaN   
1394212                         NaN                                  NaN   
1394213                         NaN                                  NaN   
1394214                         NaN                                  NaN   

         existing_loan_cnt  existing_loan_amt  
0                      4.0        162000000.0  
1                      1.0         27000000.0  
2                      5.0         15000000.0  
3                      7.0        344000000.0  
4                      1.0         16000000.0  
...                    ...                ...  
1394210                1.0                NaN  
1394211                NaN                NaN  
1394212              

In [3]:
df1.isnull().sum()

application_id                               0
user_id                                      0
birth_year                                   0
gender                                       0
insert_time                                  0
credit_score                                 0
yearly_income                                4
income_type                                  0
company_enter_month                     151881
employment_type                              0
houseown_type                                0
desired_amount                               0
purpose                                      0
personal_rehabilitation_yn              527755
personal_rehabilitation_complete_yn    1107819
existing_loan_cnt                       117650
existing_loan_amt                       229592
dtype: int64

In [4]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1277778 entries, 0 to 1394214
Data columns (total 17 columns):
 #   Column                               Non-Null Count    Dtype  
---  ------                               --------------    -----  
 0   application_id                       1277778 non-null  int64  
 1   user_id                              1277778 non-null  int64  
 2   birth_year                           1277778 non-null  float64
 3   gender                               1277778 non-null  float64
 4   insert_time                          1277778 non-null  object 
 5   credit_score                         1277778 non-null  float64
 6   yearly_income                        1277774 non-null  float64
 7   income_type                          1277778 non-null  object 
 8   company_enter_month                  1125897 non-null  float64
 9   employment_type                      1277778 non-null  object 
 10  houseown_type                        1277778 non-null  object 
 11

In [5]:
df1['company_enter_month'] = df1['company_enter_month'].astype(str)
df1['enter_month']=df1['company_enter_month'].str[:6]
df1

application_id  user_id  birth_year  gender          insert_time  \
0               1249046   118218      1985.0     1.0  2022-06-07 06:28:18   
1                954900   553686      1968.0     1.0  2022-06-07 14:29:03   
2                137274    59516      1997.0     1.0  2022-06-07 21:40:22   
3               1570936   167320      1989.0     1.0  2022-06-07 09:40:27   
4                967833    33400      2000.0     1.0  2022-06-07 08:55:07   
...                 ...      ...         ...     ...                  ...   
1394210          316356    53701      1996.0     1.0  2022-03-22 10:01:38   
1394211         1864587   489900      2000.0     1.0  2022-03-22 14:55:32   
1394212         1327066   151422      1955.0     1.0  2022-03-22 01:19:24   
1394213         1319606   173524      1983.0     1.0  2022-03-22 07:34:32   
1394214         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   

         credit_score  yearly_income      income_type company_enter_month  \
0               660.0    108000000.0  PRIVATEBUSINESS          20151101.0   
1               870.0     30000000.0  PRIVATEBUSINESS          20070201.0   
2               710.0     30000000.0       FREELANCER          20210901.0   
3               820.0     62000000.0     EARNEDINCOME          20170101.0   
4               630.0     36000000.0     EARNEDINCOME          20210901.0   
...               ...            ...              ...                 ...   
1394210         730.0     60000000.0  PRIVATEBUSINESS            202105.0   
1394211         590.0     25000000.0       FREELANCER            202106.0   
1394212         980.0     20000000.0      OTHERINCOME                 nan   
1394213         750.0     75000000.0     EARNEDINCOME            200908.0   
1394214         640.0     50000000.0     EARNEDINCOME            201705.0   

        employment_type houseown_type  desired_amount purpose  \
0                    기타            자가       1000000.0      기타   
1                   정규직        기타가족소유      30000000.0    대환대출   
2                    기타        기타가족소유      10000000.0     생활비   
3                   정규직            자가       2000000.0     생활비   
4                   정규직        기타가족소유       5000000.0     생활비   
...                 ...           ...             ...     ...   
1394210             정규직           전월세      15000000.0    사업자금   
1394211              기타        기타가족소유       5000000.0    사업자금   
1394212              기타            자가      50000000.0     생활비   
1394213             정규직            자가     100000000.0    대환대출   
1394214             정규직            자가      10000000.0    대환대출   

         personal_rehabilitation_yn  personal_rehabilitation_complete_yn  \
0                               0.0                                  NaN   
1                               0.0                                  NaN   
2                               0.0                                  NaN   
3                               0.0                                  NaN   
4                               0.0                                  0.0   
...                             ...                                  ...   
1394210                         NaN                                  NaN   
1394211                         NaN                                  NaN   
1394212                         NaN                                  NaN   
1394213                         NaN                                  NaN   
1394214                         NaN                                  NaN   

         existing_loan_cnt  existing_loan_amt enter_month  
0                      4.0        162000000.0      201511  
1                      1.0         27000000.0      200702  
2                      5.0         15000000.0      202109  
3                      7.0        344000000.0      201701  
4                      1.0         16000000.0      202109  
...                    ...                ...         ...  
1394210                1.0                NaN

In [6]:
df1['enter_month'] = df1['enter_month'].astype(float)
df1=df1.dropna(subset=['enter_month'])
df1

application_id  user_id  birth_year  gender          insert_time  \
0               1249046   118218      1985.0     1.0  2022-06-07 06:28:18   
1                954900   553686      1968.0     1.0  2022-06-07 14:29:03   
2                137274    59516      1997.0     1.0  2022-06-07 21:40:22   
3               1570936   167320      1989.0     1.0  2022-06-07 09:40:27   
4                967833    33400      2000.0     1.0  2022-06-07 08:55:07   
...                 ...      ...         ...     ...                  ...   
1394209          256365   762698      1994.0     1.0  2022-03-22 14:42:04   
1394210          316356    53701      1996.0     1.0  2022-03-22 10:01:38   
1394211         1864587   489900      2000.0     1.0  2022-03-22 14:55:32   
1394213         1319606   173524      1983.0     1.0  2022-03-22 07:34:32   
1394214         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   

         credit_score  yearly_income      income_type company_enter_month  \
0               660.0    108000000.0  PRIVATEBUSINESS          20151101.0   
1               870.0     30000000.0  PRIVATEBUSINESS          20070201.0   
2               710.0     30000000.0       FREELANCER          20210901.0   
3               820.0     62000000.0     EARNEDINCOME          20170101.0   
4               630.0     36000000.0     EARNEDINCOME          20210901.0   
...               ...            ...              ...                 ...   
1394209         640.0     12000000.0     EARNEDINCOME            202102.0   
1394210         730.0     60000000.0  PRIVATEBUSINESS            202105.0   
1394211         590.0     25000000.0       FREELANCER            202106.0   
1394213         750.0     75000000.0     EARNEDINCOME            200908.0   
1394214         640.0     50000000.0     EARNEDINCOME            201705.0   

        employment_type houseown_type  desired_amount purpose  \
0                    기타            자가       1000000.0      기타   
1                   정규직        기타가족소유      30000000.0    대환대출   
2                    기타        기타가족소유      10000000.0     생활비   
3                   정규직            자가       2000000.0     생활비   
4                   정규직        기타가족소유       5000000.0     생활비   
...                 ...           ...             ...     ...   
1394209             일용직           전월세       5000000.0    대환대출   
1394210             정규직           전월세      15000000.0    사업자금   
1394211              기타        기타가족소유       5000000.0    사업자금   
1394213             정규직            자가     100000000.0    대환대출   
1394214             정규직            자가      10000000.0    대환대출   

         personal_rehabilitation_yn  personal_rehabilitation_complete_yn  \
0                               0.0                                  NaN   
1                               0.0                                  NaN   
2                               0.0                                  NaN   
3                               0.0                                  NaN   
4                               0.0                                  0.0   
...                             ...                                  ...   
1394209                         NaN                                  NaN   
1394210                         NaN                                  NaN   
1394211                         NaN                                  NaN   
1394213                         NaN                                  NaN   
1394214                         NaN                                  NaN   

         existing_loan_cnt  existing_loan_amt  enter_month  
0                      4.0        162000000.0     201511.0  
1                      1.0         27000000.0     200702.0  
2                      5.0         15000000.0     202109.0  
3                      7.0        344000000.0     201701.0  
4                      1.0         16000000.0     202109.0  
...                    ...                ...          ...  
1394209                6.0         760

In [7]:
df1['year']=202206
df1

/var/folders/nz/160bjb610xj3r4wkpjkgnw940000gn/T/ipykernel_87339/1384796790.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['year']=202206


application_id  user_id  birth_year  gender          insert_time  \
0               1249046   118218      1985.0     1.0  2022-06-07 06:28:18   
1                954900   553686      1968.0     1.0  2022-06-07 14:29:03   
2                137274    59516      1997.0     1.0  2022-06-07 21:40:22   
3               1570936   167320      1989.0     1.0  2022-06-07 09:40:27   
4                967833    33400      2000.0     1.0  2022-06-07 08:55:07   
...                 ...      ...         ...     ...                  ...   
1394209          256365   762698      1994.0     1.0  2022-03-22 14:42:04   
1394210          316356    53701      1996.0     1.0  2022-03-22 10:01:38   
1394211         1864587   489900      2000.0     1.0  2022-03-22 14:55:32   
1394213         1319606   173524      1983.0     1.0  2022-03-22 07:34:32   
1394214         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   

         credit_score  yearly_income      income_type company_enter_month  \
0               660.0    108000000.0  PRIVATEBUSINESS          20151101.0   
1               870.0     30000000.0  PRIVATEBUSINESS          20070201.0   
2               710.0     30000000.0       FREELANCER          20210901.0   
3               820.0     62000000.0     EARNEDINCOME          20170101.0   
4               630.0     36000000.0     EARNEDINCOME          20210901.0   
...               ...            ...              ...                 ...   
1394209         640.0     12000000.0     EARNEDINCOME            202102.0   
1394210         730.0     60000000.0  PRIVATEBUSINESS            202105.0   
1394211         590.0     25000000.0       FREELANCER            202106.0   
1394213         750.0     75000000.0     EARNEDINCOME            200908.0   
1394214         640.0     50000000.0     EARNEDINCOME            201705.0   

        employment_type houseown_type  desired_amount purpose  \
0                    기타            자가       1000000.0      기타   
1                   정규직        기타가족소유      30000000.0    대환대출   
2                    기타        기타가족소유      10000000.0     생활비   
3                   정규직            자가       2000000.0     생활비   
4                   정규직        기타가족소유       5000000.0     생활비   
...                 ...           ...             ...     ...   
1394209             일용직           전월세       5000000.0    대환대출   
1394210             정규직           전월세      15000000.0    사업자금   
1394211              기타        기타가족소유       5000000.0    사업자금   
1394213             정규직            자가     100000000.0    대환대출   
1394214             정규직            자가      10000000.0    대환대출   

         personal_rehabilitation_yn  personal_rehabilitation_complete_yn  \
0                               0.0                                  NaN   
1                               0.0                                  NaN   
2                               0.0                                  NaN   
3                               0.0                                  NaN   
4                               0.0                                  0.0   
...                             ...                                  ...   
1394209                         NaN                                  NaN   
1394210                         NaN                                  NaN   
1394211                         NaN                                  NaN   
1394213                         NaN                                  NaN   
1394214                         NaN                                  NaN   

         existing_loan_cnt  existing_loan_amt  enter_month    year  
0                      4.0        162000000.0     201511.0  202206  
1                      1.0         27000000.0     200702.0  202206  
2                      5.0         15000000.0     202109.0  202206  
3                      7.0        344000000.0     201701.0  202206  
4                      1.0         16000000.0     202109.0  202206  
...                    ...                ...      

In [8]:
df1['enter_month'] = df1['enter_month'].astype(int)
df1

/var/folders/nz/160bjb610xj3r4wkpjkgnw940000gn/T/ipykernel_87339/3874855589.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['enter_month'] = df1['enter_month'].astype(int)


application_id  user_id  birth_year  gender          insert_time  \
0               1249046   118218      1985.0     1.0  2022-06-07 06:28:18   
1                954900   553686      1968.0     1.0  2022-06-07 14:29:03   
2                137274    59516      1997.0     1.0  2022-06-07 21:40:22   
3               1570936   167320      1989.0     1.0  2022-06-07 09:40:27   
4                967833    33400      2000.0     1.0  2022-06-07 08:55:07   
...                 ...      ...         ...     ...                  ...   
1394209          256365   762698      1994.0     1.0  2022-03-22 14:42:04   
1394210          316356    53701      1996.0     1.0  2022-03-22 10:01:38   
1394211         1864587   489900      2000.0     1.0  2022-03-22 14:55:32   
1394213         1319606   173524      1983.0     1.0  2022-03-22 07:34:32   
1394214         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   

         credit_score  yearly_income      income_type company_enter_month  \
0               660.0    108000000.0  PRIVATEBUSINESS          20151101.0   
1               870.0     30000000.0  PRIVATEBUSINESS          20070201.0   
2               710.0     30000000.0       FREELANCER          20210901.0   
3               820.0     62000000.0     EARNEDINCOME          20170101.0   
4               630.0     36000000.0     EARNEDINCOME          20210901.0   
...               ...            ...              ...                 ...   
1394209         640.0     12000000.0     EARNEDINCOME            202102.0   
1394210         730.0     60000000.0  PRIVATEBUSINESS            202105.0   
1394211         590.0     25000000.0       FREELANCER            202106.0   
1394213         750.0     75000000.0     EARNEDINCOME            200908.0   
1394214         640.0     50000000.0     EARNEDINCOME            201705.0   

        employment_type houseown_type  desired_amount purpose  \
0                    기타            자가       1000000.0      기타   
1                   정규직        기타가족소유      30000000.0    대환대출   
2                    기타        기타가족소유      10000000.0     생활비   
3                   정규직            자가       2000000.0     생활비   
4                   정규직        기타가족소유       5000000.0     생활비   
...                 ...           ...             ...     ...   
1394209             일용직           전월세       5000000.0    대환대출   
1394210             정규직           전월세      15000000.0    사업자금   
1394211              기타        기타가족소유       5000000.0    사업자금   
1394213             정규직            자가     100000000.0    대환대출   
1394214             정규직            자가      10000000.0    대환대출   

         personal_rehabilitation_yn  personal_rehabilitation_complete_yn  \
0                               0.0                                  NaN   
1                               0.0                                  NaN   
2                               0.0                                  NaN   
3                               0.0                                  NaN   
4                               0.0                                  0.0   
...                             ...                                  ...   
1394209                         NaN                                  NaN   
1394210                         NaN                                  NaN   
1394211                         NaN                                  NaN   
1394213                         NaN                                  NaN   
1394214                         NaN                                  NaN   

         existing_loan_cnt  existing_loan_amt  enter_month    year  
0                      4.0        162000000.0       201511  202206  
1                      1.0         27000000.0       200702  202206  
2                      5.0         15000000.0       202109  202206  
3                      7.0        344000000.0       201701  202206  
4                      1.0         16000000.0       202109  202206  
...                    ...                ...      

In [9]:
df1.isnull().sum()

application_id                              0
user_id                                     0
birth_year                                  0
gender                                      0
insert_time                                 0
credit_score                                0
yearly_income                               1
income_type                                 0
company_enter_month                         0
employment_type                             0
houseown_type                               0
desired_amount                              0
purpose                                     0
personal_rehabilitation_yn             468312
personal_rehabilitation_complete_yn    975955
existing_loan_cnt                      101386
existing_loan_amt                      196659
enter_month                                 0
year                                        0
dtype: int64

In [10]:
from datetime import datetime as dt
import time

In [11]:
df1['enter_month'] = pd.to_datetime(df1['enter_month'],format="%Y%m")
df1['year'] = pd.to_datetime(df1['year'],format="%Y%m")

/var/folders/nz/160bjb610xj3r4wkpjkgnw940000gn/T/ipykernel_87339/322463176.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['enter_month'] = pd.to_datetime(df1['enter_month'],format="%Y%m")
/var/folders/nz/160bjb610xj3r4wkpjkgnw940000gn/T/ipykernel_87339/322463176.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['year'] = pd.to_datetime(df1['year'],format="%Y%m")


In [12]:
df1['재직일수'] =df1['year'] -  df1['enter_month']

/var/folders/nz/160bjb610xj3r4wkpjkgnw940000gn/T/ipykernel_87339/2694892669.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df1['재직일수'] =df1['year'] -  df1['enter_month']


In [14]:
df1

application_id  user_id  birth_year  gender          insert_time  \
0               1249046   118218      1985.0     1.0  2022-06-07 06:28:18   
1                954900   553686      1968.0     1.0  2022-06-07 14:29:03   
2                137274    59516      1997.0     1.0  2022-06-07 21:40:22   
3               1570936   167320      1989.0     1.0  2022-06-07 09:40:27   
4                967833    33400      2000.0     1.0  2022-06-07 08:55:07   
...                 ...      ...         ...     ...                  ...   
1394209          256365   762698      1994.0     1.0  2022-03-22 14:42:04   
1394210          316356    53701      1996.0     1.0  2022-03-22 10:01:38   
1394211         1864587   489900      2000.0     1.0  2022-03-22 14:55:32   
1394213         1319606   173524      1983.0     1.0  2022-03-22 07:34:32   
1394214         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   

         credit_score  yearly_income      income_type company_enter_month  \
0               660.0    108000000.0  PRIVATEBUSINESS          20151101.0   
1               870.0     30000000.0  PRIVATEBUSINESS          20070201.0   
2               710.0     30000000.0       FREELANCER          20210901.0   
3               820.0     62000000.0     EARNEDINCOME          20170101.0   
4               630.0     36000000.0     EARNEDINCOME          20210901.0   
...               ...            ...              ...                 ...   
1394209         640.0     12000000.0     EARNEDINCOME            202102.0   
1394210         730.0     60000000.0  PRIVATEBUSINESS            202105.0   
1394211         590.0     25000000.0       FREELANCER            202106.0   
1394213         750.0     75000000.0     EARNEDINCOME            200908.0   
1394214         640.0     50000000.0     EARNEDINCOME            201705.0   

        employment_type houseown_type  desired_amount purpose  \
0                    기타            자가       1000000.0      기타   
1                   정규직        기타가족소유      30000000.0    대환대출   
2                    기타        기타가족소유      10000000.0     생활비   
3                   정규직            자가       2000000.0     생활비   
4                   정규직        기타가족소유       5000000.0     생활비   
...                 ...           ...             ...     ...   
1394209             일용직           전월세       5000000.0    대환대출   
1394210             정규직           전월세      15000000.0    사업자금   
1394211              기타        기타가족소유       5000000.0    사업자금   
1394213             정규직            자가     100000000.0    대환대출   
1394214             정규직            자가      10000000.0    대환대출   

         personal_rehabilitation_yn  personal_rehabilitation_complete_yn  \
0                               0.0                                  NaN   
1                               0.0                                  NaN   
2                               0.0                                  NaN   
3                               0.0                                  NaN   
4                               0.0                                  0.0   
...                             ...                                  ...   
1394209                         NaN                                  NaN   
1394210                         NaN                                  NaN   
1394211                         NaN                                  NaN   
1394213                         NaN                                  NaN   
1394214                         NaN                                  NaN   

         existing_loan_cnt  existing_loan_amt enter_month       year      재직일수  
0                      4.0        162000000.0  2015-11-01 2022-06-01 2404 days  
1                      1.0         27000000.0  2007-02-01 2022-06-01 5599 days  
2                      5.0         15000000.0  2021-09-01 2022-06-01  273 days  
3                      7.0        344000000.0  2017-01-01 2022-06-01 1977 days  
4                      1.0         16000000.0  2021-09-01 20

In [15]:
df1.isnull().sum()

application_id                              0
user_id                                     0
birth_year                                  0
gender                                      0
insert_time                                 0
credit_score                                0
yearly_income                               1
income_type                                 0
company_enter_month                         0
employment_type                             0
houseown_type                               0
desired_amount                              0
purpose                                     0
personal_rehabilitation_yn             468312
personal_rehabilitation_complete_yn    975955
existing_loan_cnt                      101386
existing_loan_amt                      196659
enter_month                                 0
year                                        0
재직일수                                        0
dtype: int64

In [16]:
# 자기회생여부 전처리
# 1. personal_rehabilitation_yn == 0 -> personal_rehabilitation_complete_yn = 0
# 2. personal_rehabilitation_complete_yn == 1 ->  personal_rehabilitation_yn == 1
# 3. personal_rehabilitation_yn == 0 인데 personal_rehabilitation_complete_yn == 1이면 제거 => 1번 전처리로 처리됨.

# 1
indexs_1 = df1[(df1['personal_rehabilitation_yn'] == 0 ) & (df1['personal_rehabilitation_complete_yn'] != 0)].index
df1.loc[indexs_1, ['personal_rehabilitation_complete_yn']] = 0
df1.isnull().sum()

/Users/seonghyeonma/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexing.py:1817: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(loc, value, pi)


application_id                              0
user_id                                     0
birth_year                                  0
gender                                      0
insert_time                                 0
credit_score                                0
yearly_income                               1
income_type                                 0
company_enter_month                         0
employment_type                             0
houseown_type                               0
desired_amount                              0
purpose                                     0
personal_rehabilitation_yn             468312
personal_rehabilitation_complete_yn    468312
existing_loan_cnt                      101386
existing_loan_amt                      196659
enter_month                                 0
year                                        0
재직일수                                        0
dtype: int64

In [17]:
# 2
indexs_2 = df1[(df1['personal_rehabilitation_complete_yn'] == 1 ) & (df1['personal_rehabilitation_yn'] != 1)].index
indexs_2

df1.loc[indexs_2, ['personal_rehabilitation_yn']] = 1
df1.isnull().sum()

application_id                              0
user_id                                     0
birth_year                                  0
gender                                      0
insert_time                                 0
credit_score                                0
yearly_income                               1
income_type                                 0
company_enter_month                         0
employment_type                             0
houseown_type                               0
desired_amount                              0
purpose                                     0
personal_rehabilitation_yn             468312
personal_rehabilitation_complete_yn    468312
existing_loan_cnt                      101386
existing_loan_amt                      196659
enter_month                                 0
year                                        0
재직일수                                        0
dtype: int64

In [18]:
df1[df1['personal_rehabilitation_yn'].isnull()]

application_id  user_id  birth_year  gender          insert_time  \
47              1349293   521056      1995.0     0.0  2022-06-07 16:57:35   
102              865393   302309      1991.0     0.0  2022-06-07 07:44:15   
148              591548   147098      1985.0     0.0  2022-06-07 13:21:16   
155             1617530   379488      1976.0     1.0  2022-04-20 10:08:40   
159             1232849   575347      1982.0     1.0  2022-04-20 11:49:52   
...                 ...      ...         ...     ...                  ...   
1394209          256365   762698      1994.0     1.0  2022-03-22 14:42:04   
1394210          316356    53701      1996.0     1.0  2022-03-22 10:01:38   
1394211         1864587   489900      2000.0     1.0  2022-03-22 14:55:32   
1394213         1319606   173524      1983.0     1.0  2022-03-22 07:34:32   
1394214         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   

         credit_score  yearly_income      income_type company_enter_month  \
47              560.0     30000000.0  PRIVATEBUSINESS          20200701.0   
102             620.0     36000000.0     EARNEDINCOME          20210801.0   
148             630.0     54000000.0     EARNEDINCOME          20210601.0   
155             600.0    100000000.0  PRIVATEBUSINESS            201210.0   
159             620.0     48000000.0     EARNEDINCOME            202202.0   
...               ...            ...              ...                 ...   
1394209         640.0     12000000.0     EARNEDINCOME            202102.0   
1394210         730.0     60000000.0  PRIVATEBUSINESS            202105.0   
1394211         590.0     25000000.0       FREELANCER            202106.0   
1394213         750.0     75000000.0     EARNEDINCOME            200908.0   
1394214         640.0     50000000.0     EARNEDINCOME            201705.0   

        employment_type houseown_type  desired_amount purpose  \
47                  정규직        기타가족소유      30000000.0    대환대출   
102                 정규직           전월세      38000000.0    대환대출   
148                 정규직            자가      10000000.0     생활비   
155                 정규직        기타가족소유      10000000.0     생활비   
159                 계약직           전월세       5000000.0     생활비   
...                 ...           ...             ...     ...   
1394209             일용직           전월세       5000000.0    대환대출   
1394210             정규직           전월세      15000000.0    사업자금   
1394211              기타        기타가족소유       5000000.0    사업자금   
1394213             정규직            자가     100000000.0    대환대출   
1394214             정규직            자가      10000000.0    대환대출   

         personal_rehabilitation_yn  personal_rehabilitation_complete_yn  \
47                              NaN                                  NaN   
102                             NaN                                  NaN   
148                             NaN                                  NaN   
155                             NaN                                  NaN   
159                             NaN                                  NaN   
...                             ...                                  ...   
1394209                         NaN                                  NaN   
1394210                         NaN                                  NaN   
1394211                         NaN                                  NaN   
1394213                         NaN                                  NaN   
1394214                         NaN                                  NaN   

         existing_loan_cnt  existing_loan_amt enter_month       year      재직일수  
47                     4.0         28000000.0  2020-07-01 2022-06-01  700 days  
102                    NaN                NaN  2021-08-01 2022-06-01  304 days  
148                    NaN                NaN  2021-06-01 2022-06-01  365 days  
155                    1.0         14000000.0  2012-10-01 2022-06-01 3530 days  
159                    1.0                NaN  2022-02-01 20

107896 개의 personal_rehabilitation_yn 와 personal_rehabilitation_complete_yn 이 둘다 NULL일 때는 어째야 하지<br>
다 0으로 처리해야 되는 것으로 판단하면<br>
아래 코드 실행

In [19]:
# 위에 텍스트 읽고 실행
indexs_3 = df1[(df1['personal_rehabilitation_yn'].isnull()) & (df1['personal_rehabilitation_complete_yn'].isnull()) ].index
df1.loc[indexs_3, ['personal_rehabilitation_yn']] = 0
df1.loc[indexs_3, ['personal_rehabilitation_complete_yn']] = 0

df1.isnull().sum()

/Users/seonghyeonma/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexing.py:1817: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(loc, value, pi)


application_id                              0
user_id                                     0
birth_year                                  0
gender                                      0
insert_time                                 0
credit_score                                0
yearly_income                               1
income_type                                 0
company_enter_month                         0
employment_type                             0
houseown_type                               0
desired_amount                              0
purpose                                     0
personal_rehabilitation_yn                  0
personal_rehabilitation_complete_yn         0
existing_loan_cnt                      101386
existing_loan_amt                      196659
enter_month                                 0
year                                        0
재직일수                                        0
dtype: int64

In [20]:
df1

application_id  user_id  birth_year  gender          insert_time  \
0               1249046   118218      1985.0     1.0  2022-06-07 06:28:18   
1                954900   553686      1968.0     1.0  2022-06-07 14:29:03   
2                137274    59516      1997.0     1.0  2022-06-07 21:40:22   
3               1570936   167320      1989.0     1.0  2022-06-07 09:40:27   
4                967833    33400      2000.0     1.0  2022-06-07 08:55:07   
...                 ...      ...         ...     ...                  ...   
1394209          256365   762698      1994.0     1.0  2022-03-22 14:42:04   
1394210          316356    53701      1996.0     1.0  2022-03-22 10:01:38   
1394211         1864587   489900      2000.0     1.0  2022-03-22 14:55:32   
1394213         1319606   173524      1983.0     1.0  2022-03-22 07:34:32   
1394214         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   

         credit_score  yearly_income      income_type company_enter_month  \
0               660.0    108000000.0  PRIVATEBUSINESS          20151101.0   
1               870.0     30000000.0  PRIVATEBUSINESS          20070201.0   
2               710.0     30000000.0       FREELANCER          20210901.0   
3               820.0     62000000.0     EARNEDINCOME          20170101.0   
4               630.0     36000000.0     EARNEDINCOME          20210901.0   
...               ...            ...              ...                 ...   
1394209         640.0     12000000.0     EARNEDINCOME            202102.0   
1394210         730.0     60000000.0  PRIVATEBUSINESS            202105.0   
1394211         590.0     25000000.0       FREELANCER            202106.0   
1394213         750.0     75000000.0     EARNEDINCOME            200908.0   
1394214         640.0     50000000.0     EARNEDINCOME            201705.0   

        employment_type houseown_type  desired_amount purpose  \
0                    기타            자가       1000000.0      기타   
1                   정규직        기타가족소유      30000000.0    대환대출   
2                    기타        기타가족소유      10000000.0     생활비   
3                   정규직            자가       2000000.0     생활비   
4                   정규직        기타가족소유       5000000.0     생활비   
...                 ...           ...             ...     ...   
1394209             일용직           전월세       5000000.0    대환대출   
1394210             정규직           전월세      15000000.0    사업자금   
1394211              기타        기타가족소유       5000000.0    사업자금   
1394213             정규직            자가     100000000.0    대환대출   
1394214             정규직            자가      10000000.0    대환대출   

         personal_rehabilitation_yn  personal_rehabilitation_complete_yn  \
0                               0.0                                  0.0   
1                               0.0                                  0.0   
2                               0.0                                  0.0   
3                               0.0                                  0.0   
4                               0.0                                  0.0   
...                             ...                                  ...   
1394209                         0.0                                  0.0   
1394210                         0.0                                  0.0   
1394211                         0.0                                  0.0   
1394213                         0.0                                  0.0   
1394214                         0.0                                  0.0   

         existing_loan_cnt  existing_loan_amt enter_month       year      재직일수  
0                      4.0        162000000.0  2015-11-01 2022-06-01 2404 days  
1                      1.0         27000000.0  2007-02-01 2022-06-01 5599 days  
2                      5.0         15000000.0  2021-09-01 2022-06-01  273 days  
3                      7.0        344000000.0  2017-01-01 2022-06-01 1977 days  
4                      1.0         16000000.0  2021-09-01 20

In [21]:
purpose = {'purpose' : {'ETC':'기타', 'INVEST':'투자','LIVING':'생활비','BUSINESS':'사업자금','BUYCAR':'자동차구입','BUYHOUSE':'주택구입','SWITCHLOAN':'대환대출','HOUSEDEPOSIT':'전월세보증금'}}
df1 = df1.replace(purpose)
df1['purpose'].unique()

array(['기타', '대환대출', '생활비', '사업자금', '주택구입', '전월세보증금', '투자', '자동차구입'],
      dtype=object)

In [54]:
#df1.to_csv('user_spec_clean.csv')

In [3]:
df2 = pd.read_csv('data/loan_result.csv')
df2

application_id loanapply_insert_time  bank_id  product_id  \
0                1748340   2022-06-07 13:05:41        7         191   
1                1748340   2022-06-07 13:05:41       25         169   
2                1748340   2022-06-07 13:05:41        2           7   
3                1748340   2022-06-07 13:05:41        4         268   
4                1748340   2022-06-07 13:05:41       11         118   
...                  ...                   ...      ...         ...   
13527358         1428218   2022-06-03 12:01:58       62         200   
13527359         1428218   2022-06-03 12:01:49        2           7   
13527360         1428218   2022-06-03 12:01:48       32         257   
13527361         1428218   2022-06-03 12:01:48       33         110   
13527362         1428218   2022-06-03 12:01:48        5         194   

          loan_limit  loan_rate  is_applied  
0         42000000.0       13.6         NaN  
1         24000000.0       17.9         NaN  
2         24000000.0       18.5         NaN  
3         29000000.0       10.8         NaN  
4          5000000.0       16.4         NaN  
...              ...        ...         ...  
13527358   3000000.0       14.8         NaN  
13527359  40000000.0       11.8         NaN  
13527360  15000000.0        7.2         NaN  
13527361  44000000.0       13.5         NaN  
13527362  44000000.0        9.7         NaN  

[13527363 rows x 7 columns]

In [23]:
#승인한도/금리제거하기
df2=df2.dropna(subset=['loan_limit'])
df2=df2.dropna(subset=['loan_rate'])
df2.isnull().sum()

application_id                 0
loanapply_insert_time          0
bank_id                        0
product_id                     0
loan_limit                     0
loan_rate                      0
is_applied               3255482
dtype: int64

In [53]:
#df2.to_csv('loan_result_clean.csv')

In [24]:
#분석
from sklearn.model_selection import train_test_split

In [25]:
#데이터병합
df3 = pd.merge(df1, df2)
df3

application_id  user_id  birth_year  gender          insert_time  \
0                 954900   553686      1968.0     1.0  2022-06-07 14:29:03   
1                 954900   553686      1968.0     1.0  2022-06-07 14:29:03   
2                 954900   553686      1968.0     1.0  2022-06-07 14:29:03   
3                 954900   553686      1968.0     1.0  2022-06-07 14:29:03   
4                 954900   553686      1968.0     1.0  2022-06-07 14:29:03   
...                  ...      ...         ...     ...                  ...   
11565277         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   
11565278         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   
11565279         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   
11565280         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   
11565281         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   

          credit_score  yearly_income      income_type company_enter_month  \
0                870.0     30000000.0  PRIVATEBUSINESS          20070201.0   
1                870.0     30000000.0  PRIVATEBUSINESS          20070201.0   
2                870.0     30000000.0  PRIVATEBUSINESS          20070201.0   
3                870.0     30000000.0  PRIVATEBUSINESS          20070201.0   
4                870.0     30000000.0  PRIVATEBUSINESS          20070201.0   
...                ...            ...              ...                 ...   
11565277         640.0     50000000.0     EARNEDINCOME            201705.0   
11565278         640.0     50000000.0     EARNEDINCOME            201705.0   
11565279         640.0     50000000.0     EARNEDINCOME            201705.0   
11565280         640.0     50000000.0     EARNEDINCOME            201705.0   
11565281         640.0     50000000.0     EARNEDINCOME            201705.0   

         employment_type  ... existing_loan_amt  enter_month       year  \
0                    정규직  ...        27000000.0   2007-02-01 2022-06-01   
1                    정규직  ...        27000000.0   2007-02-01 2022-06-01   
2                    정규직  ...        27000000.0   2007-02-01 2022-06-01   
3                    정규직  ...        27000000.0   2007-02-01 2022-06-01   
4                    정규직  ...        27000000.0   2007-02-01 2022-06-01   
...                  ...  ...               ...          ...        ...   
11565277             정규직  ...       117000000.0   2017-05-01 2022-06-01   
11565278             정규직  ...       117000000.0   2017-05-01 2022-06-01   
11565279             정규직  ...       117000000.0   2017-05-01 2022-06-01   
11565280             정규직  ...       117000000.0   2017-05-01 2022-06-01   
11565281             정규직  ...       117000000.0   2017-05-01 2022-06-01   

              재직일수  loanapply_insert_time  bank_id  product_id  loan_limit  \
0        5599 days    2022-06-07 14:29:28       29          20  50000000.0   
1        5599 days    2022-06-07 14:29:05       14         128  20000000.0   
2        5599 days    2022-06-07 14:29:23       34         215   5000000.0   
3        5599 days    2022-06-07 14:29:05       32          56  15000000.0   
4        5599 days    2022-06-07 14:29:26       55         186  43000000.0   
...            ...                    ...      ...         ...         ...   
11565277 1857 days    2022-03-22 22:12:38       37         206  17000000.0   
11565278 1857 days    2022-03-22 22:12:36       19         231  20000000.0   
11565279 1857 days    2022-03-22 22:12:37       33         110  10000000.0   
11565280 1857 days    2022-03-22 22:12:36       50         142  20000000.0   
11565281 1857 days    2022-03-22 22:12:38       56           5  10000000.0   

         loan_rate is_applied  
0              6.9        NaN  
1              6.9        NaN  
2             14.9        NaN  
3             13.9        NaN  
4             11.6        NaN  
...            ...        ...  
11565277      19.9        0.0  
11565278      18.9        0.0  
11565279  

In [46]:
df3['is_applied'].unique()

array([nan,  0.,  1.])

In [47]:
df3['is_applied'].isnull().sum()

2874091

In [48]:
df3=df3.dropna(subset=['is_applied'])

In [49]:
df3.tail(5)

application_id  user_id  birth_year  gender          insert_time  \
11565277         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   
11565278         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   
11565279         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   
11565280         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   
11565281         1482466   766546      1975.0     1.0  2022-03-22 22:12:35   

          credit_score  yearly_income   income_type company_enter_month  \
11565277         640.0     50000000.0  EARNEDINCOME            201705.0   
11565278         640.0     50000000.0  EARNEDINCOME            201705.0   
11565279         640.0     50000000.0  EARNEDINCOME            201705.0   
11565280         640.0     50000000.0  EARNEDINCOME            201705.0   
11565281         640.0     50000000.0  EARNEDINCOME            201705.0   

         employment_type  ... existing_loan_amt  enter_month       year  \
11565277             정규직  ...       117000000.0   2017-05-01 2022-06-01   
11565278             정규직  ...       117000000.0   2017-05-01 2022-06-01   
11565279             정규직  ...       117000000.0   2017-05-01 2022-06-01   
11565280             정규직  ...       117000000.0   2017-05-01 2022-06-01   
11565281             정규직  ...       117000000.0   2017-05-01 2022-06-01   

              재직일수  loanapply_insert_time  bank_id  product_id  loan_limit  \
11565277 1857 days    2022-03-22 22:12:38       37         206  17000000.0   
11565278 1857 days    2022-03-22 22:12:36       19         231  20000000.0   
11565279 1857 days    2022-03-22 22:12:37       33         110  10000000.0   
11565280 1857 days    2022-03-22 22:12:36       50         142  20000000.0   
11565281 1857 days    2022-03-22 22:12:38       56           5  10000000.0   

         loan_rate is_applied  
11565277      19.9        0.0  
11565278      18.9        0.0  
11565279      13.3        0.0  
11565280      17.8        0.0  
11565281      16.3        0.0  

[5 rows x 26 columns]

In [50]:
#is_applied 형식 변환
df3['is_applied'] = df3['is_applied'].astype(str)
df3['is_applied'].unique()

/var/folders/nz/160bjb610xj3r4wkpjkgnw940000gn/T/ipykernel_87339/1210094961.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df3['is_applied'] = df3['is_applied'].astype(str)


array(['0.0', '1.0'], dtype=object)

In [52]:
applied = {'is_applied' : {'0.0':0,'1.0':1}}
df3 = df3.replace(applied)
df3['is_applied'].unique()

array([0, 1])

In [53]:
df3.isnull().sum()

application_id                               0
user_id                                      0
birth_year                                   0
gender                                       0
insert_time                                  0
credit_score                                 0
yearly_income                                0
income_type                                  0
company_enter_month                          0
employment_type                              0
houseown_type                                0
desired_amount                               0
purpose                                      0
personal_rehabilitation_yn                   0
personal_rehabilitation_complete_yn          0
existing_loan_cnt                      1133896
existing_loan_amt                      1972707
enter_month                                  0
year                                         0
재직일수                                         0
loanapply_insert_time                        0
bank_id      

In [56]:
#input 변수와 Output 변수 구분 
X=np.array(df3.iloc[:,1:])
y=np.array(df3['is_applied'])

In [57]:
X

array([[836762, 1979.0, 1.0, ..., 3000000.0, 14.5, 0],
       [836762, 1979.0, 1.0, ..., 1000000.0, 19.9, 0],
       [186886, 1979.0, 1.0, ..., 11000000.0, 15.1, 0],
       ...,
       [766546, 1975.0, 1.0, ..., 10000000.0, 13.3, 0],
       [766546, 1975.0, 1.0, ..., 20000000.0, 17.8, 0],
       [766546, 1975.0, 1.0, ..., 10000000.0, 16.3, 0]], dtype=object)

In [58]:
y

array([0, 0, 0, ..., 0, 0, 0])

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=0.3,random_state=0)

print("X_train 크기:", X_train.shape)
print("y_train 크기:", y_train.shape)
print("X_test 크기:", X_test.shape)
print("y_test 크기:", y_test.shape)

In [ ]:
#칼럼별 고유값 확인
for col in X_train.columns:
  cols = str(col)
  print(f'{col} 고유값 : {X_train[cols].unique()} \n')

- 명목형 데이터:
- 이산형 데이터: 
- 순서형 데이터:
- 이진형 데이터: 

In [ ]:
#타겟 비율 확인하기
import matplotlib as mpl

In [ ]:
#대출 신청 여부의 비율을 확인하기 위한 그래프
def write_percent(ax,total_size):
    for patch in ax.patches:
        height = patch.get_height()
        width= patch.get_width()
        percent= height/total_size*100 #타깃값 비율

mpl.rc('font',size=15)
plt.figure(figsize(8,7))
ax= sns.countplot(x='is_applied',data=X_train)
write_percent(ax,len(X_train))
ax.set_title('target distribution')